### PR TITLE
Extrinsics async (part 1)

### PR DESCRIPTION
**Pull request by basfroman**
Async migration for the folowing modules:

- `bittensor/extrinsics/commit_weights.py`
- `bittensor/extrinsics/delegation.py`

For easy of review, each function is in a separate commit.

### CODE REVIEW

File: bittensor/extrinsics/delegation.py [hunk 1]
```diff
@@ -126,19 +125,19 @@ def delegate_extrinsic(
...
         raise NotDelegateError("Hotkey: {} is not a delegate.".format(delegate_ss58))
 
     # Get state.
-    my_prev_coldkey_balance = subtensor.get_balance(wallet.coldkey.ss58_address)
-    delegate_take = subtensor.get_delegate_take(delegate_ss58)
-    delegate_owner = subtensor.get_hotkey_owner(delegate_ss58)
-    my_prev_delegated_stake = subtensor.get_stake_for_coldkey_and_hotkey(
+    my_prev_coldkey_balance = await subtensor.get_balance(wallet.coldkey.ss58_address)
+    delegate_owner = await subtensor.get_hotkey_owner(delegate_ss58)
+    my_prev_delegated_stake = await subtensor.get_stake_for_coldkey_and_hotkey(
```
Comment (opendansor) [inline bittensor/extrinsics/delegation.py]: do these need to be gathered before moving on?

**Review by opendansor — APPROVED** [review]

**Review by ibraheem-abe — APPROVED** [review]
Thanks for answering my queries

File: bittensor/extrinsics/delegation.py (hunk 1 above)
Comment (basfroman) [reply bittensor/extrinsics/delegation.py]: I thought about migrating first, and then apply asincoo.gather during the optimization + testing. Do you think it's worth it right away?

**Comment by basfroman** [comment]
> Thanks for answering my queries

Always welcome!

File: bittensor/extrinsics/delegation.py (hunk 1 above)
Comment (opendansor) [reply bittensor/extrinsics/delegation.py]: Thats fine! Thanks for clearing up your thought process.

**Pull request merged**